src/anomalib/post_processing/base.py

```diff
@@ -0,0 +1,19 @@
+"""Base class for post-processor."""
+
+from abc import ABC, abstractmethod
+
+from lightning.pytorch import Callback
+from torch import nn
+
+from anomalib.dataclasses import InferenceBatch
+
+
+class PostProcessor(nn.Module, Callback, ABC):
```

Comment:
Would it be an idea to inherit `LightningModule` instead of `nn.Module` and `Callback`? 

As far as I can see, the main reason for using `Callback` is to access the model hooks, which `LightningModule` already provides. This way it would logically make more sense since, theoretically speaking `PostProcessor` is actually a "head" module on top of an anomaly detector module.

Reply:
This would not work, since the lightning trainer only calls the model hooks of the main `LightningModule`, not those of any submodules. The only way to have the post-processor act as a head module would be to explicitly call the post-processor in the appropriate model hooks in the model implementation (usually after every test or predict step). But this would limit the flexibility of the design, because the post-processing logic may differ between different subclasses of `PostProcessor`. 

For example, some post-processors (like our standard `OneClassPostProcessor`) may need to collect statistics over the validation set, which may not be required for the post-processor of other learning types (e.g. VLM-based zero-shot approaches). With the callback design, we don't need to modify the validation hook of the `AnomalyModule`, because the `PostProcessor` itself has control over in which stages of the pipeline it operates. This improves flexibility (we can use any post-processor on any model) and reduces boilerplate code (the model implementer does not have to make any specific calls to the post-processor).